server/chat-plugins/info.js

```diff
@@ -299,6 +299,13 @@ const commands = {
 		this.sendReplyBox(buf);
 	},
 
+	sp: 'showpunishments',
+	showpunishments(target, room, user) {
+		if (room.id.includes('-')) return this.errorReply('This command can only be used in public, hidden or private rooms.');
```

Comment:
I would check for room.chatRoomData personally to check whether it's a temp room. Also, put this in the code for the page rendering as well.